pkg/domain/infra/abi/play.go

```diff
@@ -486,7 +486,7 @@ func (ic *ContainerEngine) getImageAndLabelInfo(ctx context.Context, cwd string,
 	if err != nil {
 		return nil, nil, err
 	}
-	if (len(buildFile) > 0 && !existsLocally) || (len(buildFile) > 0 && options.Build) {
+	if options.Build && ((len(buildFile) > 0 && !existsLocally) || (len(buildFile) > 0)) {
```

Comment:
The `(len(buildFile) > 0 && !existsLocally)` is unnecessary because you also check for only `(len(buildFile) > 0)`

Reply:
Fixed.

> I think the problem with this is that the image is build every time. The current behaviour is build when image not exists now it will build every time. I don't think this is a good user experience.

@Luap99 For this i think previous behavior was wrong, we should never try to `build` if `--build` is not set. I am fine to move back default value of `--build=false` but it changes behavior. Hence to be as close as to the previous behviour i am setting default value of build to `true`.

Reply:
But this changes behaviour, if I have a bigger Containerfile and it start building every time this will be super annoying.
IMO it should be:
`--build` or `--build=true` means always build.
when `--build` is not set it means only build when image not found locally
`--build=false` should mean never build

You can check if a flag is set on the cli with flags.Changed("build")

Reply:
@Luap99 I am still confused why do we want to build if user does not specifies `--build` but sure I agree lets keep the old behavior intact. I think recent commit should addresses this.